sparkler-app/src/main/scala/edu/usc/irds/sparkler/solr/SolrUpsert.scala

```diff
@@ -38,9 +37,16 @@ class SolrUpsert(job: SparklerJob) extends ((TaskContext, Iterator[Resource]) =>
     val solrClient = job.newCrawlDbSolrClient()
 
     //TODO: handle this in server side - tell solr to skip docs if they already exist
-    val newResources: Iterator[Resource] = for (doc <- docs if solrClient.crawlDb.getById(doc.getId) == null) yield doc
+    var newResources: Set[Resource] = Set()
```

Comment:
I am gonna restore lazy evaluations here.

FYI, the catch here is when we convert `Iterator[Resource]` to `Set[resource]` lazy evaluations are gone and thus we are starting to buffer, eventually it may overflow in some settings!